services/navigation/builtin/builtin.go

```diff
@@ -93,16 +98,26 @@ func (conf *Config) Validate(path string) ([]string, error) {
 	}
 	deps = append(deps, conf.BaseName)
 
-	if conf.MovementSensorName == "" {
-		return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
-	}
-	deps = append(deps, conf.MovementSensorName)
-
 	if conf.MotionServiceName == "" {
 		conf.MotionServiceName = resource.DefaultServiceName
 	}
 	deps = append(deps, resource.NewName(motion.API, conf.MotionServiceName).String())
 
+	if conf.MapType == "" {
+		conf.MapType = defaultMapType
+	}
+
+	if !slices.Contains([]string{"None", "GPS"}, conf.MapType) {
+		return nil, errors.New("invalid map_type, when defined map_type must be one of the following ['None', 'GPS']")
+	}
+
+	if conf.MapType == "GPS" {
+		if conf.MovementSensorName == "" {
+			return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
+		}
+		deps = append(deps, conf.MovementSensorName)
+	}
+
```

Comment:
Could this be simplified by using a switch?
```
switch conf.MapTypeName {
case "":
   ...
case "GPS":
   ...
case "None":
   ...
case default:
   return nil, errors.New("invalid map_type, when defined map_type must be one of the following ['None', 'GPS']")
}
```

Reply:
I feel that the way I have it setup provides additional context to each check. I am sometimes against using a switch statement when each case doing a different thing. However, if the switch is still preferred I'm happy to change to it.

Reply:
Spoke with Kim a bit more offline and we think this question is related to what validation checks should be done in `Validate()` vs. `New()`/`Reconfigure()`. 

Should validate only look for the presences or absences of required components and assign default values?

- If so, I will move the checks of `map_type`/`movement_sensor` to `Reconfigure()`, leaving only the setting to default in `Validate()`
- If not we can keep it as is with the format of handling each config attribute independently (when possible) OR go with Kim's switch statement which handles `movement_sensor` + `map_type` checks together.

Reply:
Spoke offline with Benji and Peter we have agreed on the paradigm: Validate will check if the values of attributes are potentially valid and reconfigure will handle the setting of defaults